### PR TITLE
Add support for a drawings screen video background

### PR DESCRIPTION
**Pull request by peppy**
Was requested via email from a tournament manager.

### CODE REVIEW

**Comment by peppy** [comment]
Actually hold off on this momentarily, going to do a pass on `IProvideVideo` as things are a bit all over the place right now.

**Comment by peppy** [comment]
I've changed the way fallback logic for videos works to be a lot more inclusive, and fix the remaining screens which were opting out of the video interface to provide fallback logic. See 73e924479fb6db7b5e57413ef08663c068805f3c for more details.

File: osu.Game.Tournament/TournamentSceneManager.cs [hunk 1]
```diff
@@ -186,7 +187,7 @@ public void SetScreen(Type screenType)
             var lastScreen = currentScreen;
             currentScreen = target;
 
-            if (currentScreen is IProvideVideo)
+            if (currentScreen.ChildrenOfType<TourneyVideo>().FirstOrDefault()?.VideoAvailable == true)
```
Comment (bdach) [inline osu.Game.Tournament/TournamentSceneManager.cs:190]: A few things make me wince here, namely the `.ChildrenOfType<>()` usage outside of tests, and the fact that the value of `VideoAvailable` is dependent on timing. But it doesn't look like either are going to be huge problem, especially in the latter case where the surrounding code basically guarantees a sync load (see `TournamentSceneManager`'s BDL). So I'm willing to look away, but I'm probably going to let this hang a bit and see if anyone finds any of the above offensive enough to request changes.

**Review by bdach — DISMISSED** [review]

File: osu.Game.Tournament/TournamentSceneManager.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Tournament/TournamentSceneManager.cs:190]: I initially added a `bool` to the interface but it just felt like way too much implementation (was touching every screen, adding new fields, then mapping to the property) to do something that can be done in a single line. And had more chance of going wrong (screens missing the interface, implementing wrong etc.)

Comment (bdach) [reply osu.Game.Tournament/TournamentSceneManager.cs:190]: `ChildrenOfType` is the lesser of the two mentioned problems for me.

I'm just going to go ahead and merge without spending too much time on this. It's tournament client.

File: osu.Game.Tournament/Screens/Setup/SetupScreen.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 namespace osu.Game.Tournament.Screens.Setup
 {
-    public class SetupScreen : TournamentScreen, IProvideVideo
+    public class SetupScreen : TournamentScreen
```
Comment (frenzibyte) [inline osu.Game.Tournament/Screens/Setup/SetupScreen.cs:22]: Not sure how big of a deal this is, but the setup screen was (in a dodgy way) forcing the main video to hide by inheriting `IProvideVideo` despite not having any actual video background sprite.

With this PR, now the main video shows on setup screen.

Comment (bdach) [reply osu.Game.Tournament/Screens/Setup/SetupScreen.cs:22]: Well okay, this is probably a bit bad. Disabling automerge for now.

Comment (peppy) [reply osu.Game.Tournament/Screens/Setup/SetupScreen.cs:24]: I knew this was the case but didn't really think too much of it. I've added a box to hide it.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**